core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraImpl.kt

```diff
@@ -31,7 +33,7 @@ class TrackChunkDescriptor(
     val curves: DirectionalMap<DistanceRangeMap<Double>>,
     val gradients: DirectionalMap<DistanceRangeMap<Double>>,
     val length: Distance,
-    val routes: DirectionalMap<StaticIdxCollection<Route>>,
```

Comment:
Can you recall why ordering of routes per track chunk is needed?

Reply:
It's not about the ordering, it's about kotlin/java interface. We couldn't get the typing to work properly when directly iterating over kotlin collections, it would type elements as `StaticIdx` instead of `int`, which would fail to compile. Accessing elements with `.get` does work, but it can't be done on unordered collections.